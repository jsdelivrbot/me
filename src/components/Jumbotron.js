import React from 'react';

import '../styles/components/Jumbotron.css';

const Jumbotron = props => {
  const {
    headline,
    subhead
  } = props;
  return (
    <div className="Jumbotron text-white">
      {headline &&
        <h2 className="App-title">{ headline }</h2>
      }
      {subhead &&
        <p className="lead">{ subhead }</p>
      }
    </div>
  );
};

export default Jumbotron;