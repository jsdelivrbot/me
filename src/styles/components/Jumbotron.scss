@import '../includes';

.Jumbotron {
  background: rgba(69, 39, 160, 1);
  background: -moz-linear-gradient(45deg, rgba(69, 39, 160, 1) 0%, rgba(113, 30, 155, 1) 100%);
  background: -webkit-gradient(left bottom, right top, color-stop(0%, rgba(69, 39, 160, 1)), color-stop(100%, rgba(113, 30, 155, 1)));
  background: -webkit-linear-gradient(45deg, rgba(69, 39, 160, 1) 0%, rgba(113, 30, 155, 1) 100%);
  background: -o-linear-gradient(45deg, rgba(69, 39, 160, 1) 0%, rgba(113, 30, 155, 1) 100%);
  background: -ms-linear-gradient(45deg, rgba(69, 39, 160, 1) 0%, rgba(113, 30, 155, 1) 100%);
  background: linear-gradient(45deg, rgba(69, 39, 160, 1) 0%, rgba(113, 30, 155, 1) 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#4527a0', endColorstr='#711e9b', GradientType=1 );
}

.Jumbotron h2 {
  font-family: Quicksand,sans-serif;
  font-size: 2.124em;

  @media only screen and (min-width: $lg) {
    font-size: 3.124em;
  }
}