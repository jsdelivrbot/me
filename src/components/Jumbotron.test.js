import React from 'react';
import renderer from 'react-test-renderer';

import Jumbotron from './Jumbotron';

describe('Jumbotron', () => {
  const makeComponent = customProps => {
    const defaultProps = {
      headline: 'Foo',
      subhead: 'Bar'
    };
    const props = { ...defaultProps, ...customProps };
    return renderer.create(
      <Jumbotron { ...props } />,
    );
  };

  test('_SNAPSHOT_', () => {
    const component = makeComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const component = makeComponent();
    expect(component).toBe.ok;
  });
});