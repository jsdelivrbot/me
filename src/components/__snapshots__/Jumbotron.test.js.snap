// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jumbotron _SNAPSHOT_ 1`] = `
<div
  className="Jumbotron text-white"
>
  <h2
    className="App-title"
  >
    Foo
  </h2>
  <p
    className="lead"
  >
    Bar
  </p>
</div>
`;
